src/Assembly/Helpers/Net/Sockets/SocketStream.cs

```
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Assembly.Helpers.Net.Sockets
+{
+    public class SocketStream : Stream
+    {
+        private IPokeCommandStarter _starter;
+        private List<SocketAction> _actions;
+        private string _buildName;
+        private string _cacheName;
+
+        public SocketStream(IPokeCommandStarter starter, string buildName, string cacheName)
+        {
+            _starter = starter;
+            _actions = new List<SocketAction>();
+            _buildName = buildName;
+            _cacheName = cacheName;
+        }
+
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            switch (origin)
+            {
+                case SeekOrigin.Begin:
+                    Position = offset;
+                    break;
+
+                case SeekOrigin.Current:
+                    Position += offset;
+                    break;
+
+                case SeekOrigin.End:
+                    Position = 0x100000000 - offset;
```

This is broken - `SeekOrigin.End` doesn't mean "backwards from the end". I don't see any reason to seek to the end of a stream like this, so maybe we should just throw an exception here?